File: games/lia-1/bots/java/src/core/api/Response.java
package core.api;

import java.util.ArrayList;

public class Response {
    public int __uid;
    private ArrayList<Command> commands = new ArrayList<>();

    /** Change thrust speed of a unit */
    public void setSpeed(int unitId, Speed speed) {
        commands.add(new SpeedEvent(unitId, speed));
    }

    /** Change rotation speed of a unit */
    public void setRotation(int unitId, Rotation rotation) {
        commands.add(new RotationEvent(unitId, rotation));
    }

    /** Make a unit shoot */
    public void shoot(int unitId) {
        commands.add(new ShootEvent(unitId));
    }

    /** Start navigation */
    public void navigationStart(int unitId, float x, float y, boolean moveBackwards) {
        commands.add(new NavigationStartEvent(unitId, x, y, moveBackwards));
    }

    public void navigationStart(int unitId, float x, float y) {
        commands.add(new NavigationStartEvent(unitId, x, y, false));
    }

    /** Stop navigation */
    public void navigationStop(int unitId) {
        commands.add(new NavigationStopEvent(unitId));
    }

    /** Make your unit say something */
    public void saySomething(int unitId, String text) {
        commands.add(new SaySomethingEvent(unitId, text));
    }

    public void spawnUnit(UnitType type) {
        commands.add(new SpawnUnitEvent(type));
    }

    public void addCommand(Command command) {
        commands.add(command);
    }
}

File: games/lia-1/bots/kotlin/src/core/api/Response.kt
package core.api

import java.util.ArrayList

interface Command

class Response {
    var __uid: Int = 0
    private val commands = ArrayList<Command>()

    /** Change thrust speed of a unit  */
    fun setSpeed(unitId: Int, speed: Speed) {
        commands.add(SpeedEvent(unitId, speed))
    }

    /** Change rotation speed of a unit  */
    fun setRotation(unitId: Int, rotation: Rotation) {
        commands.add(RotationEvent(unitId, rotation))
    }

    /** Make a unit shoot  */
    fun shoot(unitId: Int) {
        commands.add(ShootEvent(unitId))
    }

    /** Start navigation  */
    fun navigationStart(unitId: Int, x: Float, y: Float, moveBackwards: Boolean) {
        commands.add(NavigationStartEvent(unitId, x, y, moveBackwards))
    }

    fun navigationStart(unitId: Int, x: Float, y: Float) {
        commands.add(NavigationStartEvent(unitId, x, y, false))
    }

    /** Stop navigation  */
    fun navigationStop(unitId: Int) {
        commands.add(NavigationStopEvent(unitId))
    }

    /** Make your unit say something  */
    fun saySomething(unitId: Int, text: String) {
        commands.add(SaySomethingEvent(unitId, text))
    }

    fun spawnUnit(type: UnitType) {
        commands.add(SpawnUnitEvent(type))
    }
}

class SpeedEvent(var unitId: Int, var speed: Speed) : Command {
    var __type = SpeedEvent::class.java.simpleName
}

class SpawnUnitEvent(var type: UnitType) : Command {
    var __type = SpawnUnitEvent::class.java.simpleName
}

class ShootEvent(var unitId: Int) : Command {
    var __type = ShootEvent::class.java.simpleName
}

class RotationEvent(var unitId: Int, var rotation: Rotation) : Command {
    var __type = RotationEvent::class.java.simpleName
}

class SaySomethingEvent(var unitId: Int, var text: String) : Command {
    var __type = SaySomethingEvent::class.java.simpleName
}

class NavigationStopEvent(var unitId: Int) : Command {
    var __type = NavigationStopEvent::class.java.simpleName
}

class NavigationStartEvent(var unitId: Int, var x: Float, var y: Float, var moveBackwards: Boolean) : Command {
    var __type = NavigationStartEvent::class.java.simpleName
}